File: Leetcode/typescript/1653-min-deletions.ts
function minimumDeletions(s: string): number {
  const n = s.length;
  let aCountRight = 0;

  for (let c of s) {
    if (c === "a") {
      aCountRight++;
    }
  }
  let bCountLeft = 0;
  let res = n;
  for (let i = 0; i < n; i++) {
    if (s[i] === "a") {
      aCountRight--;
    }
    res = Math.min(res, bCountLeft + aCountRight);
    if (s[i] === "b") {
      bCountLeft += 1;
    }
  }
  return res;
}

File: Leetcode/python/1653-min-deletions.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        n = len(s)
        aCountRight = 0

        for c in s:
            aCountRight += 1 if c == "a" else 0

        bCountLeft = 0
        res = n
        for i, c in enumerate(s):
            if c == 'a':
                aCountRight -= 1
            res = min(res, bCountLeft + aCountRight)
            if c == 'b':
                bCountLeft += 1
        return res
